### PR TITLE
[#9] [Feature] As a developer, I want to use the Activity Extension

### PR DESCRIPTION
**Pull request by hoangnguyen92dn**
https://github.com/nimblehq/android-common-ktx/issues/9

## What happened 👀

Kotlin provides the ability to extend a class with new functionality without having to inherit from the class or use design patterns such as Decorator. This is done via special declarations called extensions.

## Insight 📝

Add the useful extensions from the Activity class for using:

- Check/Request Permission, Navigation result, Change status bar color
- Add Unit Test

## Proof Of Work 📹

N/A

### CODE REVIEW

**Comment by doannimble** [comment]
@hoangnguyen92dn Can we add more `permission` extension?

**Comment by hoangnguyen92dn** [comment]
@doannimble Sure. I will collect more common extensions and add to this PR

**Comment by hoangnguyen92dn** [comment]
@doannimble @lydiasama As discussed in the meeting, we decided to get rid of the extension for permission 🙏

File: common-ktx/src/main/java/co/nimblehq/common/extensions/ActivityExt.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Runtime.getRuntime().exit(0)
+}
+
+/**
+ * Change the status bar color.
+ *
+ * @param colorResId Color resource id
+ */
+fun Activity.changeStatusBarColor(@ColorRes colorResId: Int) {
+    val window = this.window
```
Comment (doannimble) [inline common-ktx/src/main/java/co/nimblehq/common/extensions/ActivityExt.kt:52]: @hoangnguyen92dn Can we use  with`with(window)` ?

**Review by doannimble — COMMENTED** [review]
Rest LGTM  👍

File: common-ktx/src/main/java/co/nimblehq/common/extensions/ActivityExt.kt (hunk 1 above)
Comment (hoangnguyen92dn) [reply common-ktx/src/main/java/co/nimblehq/common/extensions/ActivityExt.kt:52]: @minhnimble @lydiasama I tried to apply `with` but AS shows redundant
image

**Comment by lydiasama** [comment]
Screen Shot 2021-11-23 at 9 53 39 PM
@hoangnguyen92dn When you don't want to keep the keyboard showing across the application when you minimize your app. 😄

**Comment by hoangnguyen92dn** [comment]
@lydiasama Sounds good to me 👍  I will add it in.

**Comment by hoangnguyen92dn** [comment]
@lydiasama I double-checked and see that `KeyboardListener` is the custom class. I thought it is from the Android SDK package, so I think we could get rid of it from the extension.

**Review by doannimble — APPROVED** [review]
Good to go 🏃

**Pull request merged**